components/sts/org.wso2.carbon.identity.sts.passive/src/main/java/org/wso2/carbon/identity/sts/passive/processors/RequestProcessor.java

```diff
@@ -138,10 +152,30 @@ protected SAMLTokenIssuerConfig getSAMLTokenIssuerConfig(AxisService service, bo
             return null;
         }
 
+        tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        try {
+            identityProvider = IdentityProviderManager.getInstance().getResidentIdP(tenantDomain);
+        } catch (Exception e) {
+            throw new Exception(
+                    "Error occurred while retrieving Resident Identity Provider information for tenant " + tenantDomain,
+                    e);
+        }
+
+        FederatedAuthenticatorConfig[] authnConfigs = identityProvider.getFederatedAuthenticatorConfigs();
+        for (FederatedAuthenticatorConfig config : authnConfigs) {
```

Comment:
Use IdentityApplicationManagementUtil.getFederatedAuthenticator instead of looping through.